### PR TITLE
Update Node.js dependencies

### PR DESCRIPTION
**Pull request by lb-**
- General update for packages to keep us on the latest versions.
- Includes updates for browserslist and any npm audit fixes.
- Excludes Stylelint - waiting for https://github.com/wagtail/stylelint-config-wagtail/pull/34 to be released
- Excludes the autocomplete package as that needs a bit more testing
- All other major version updates are Node.js or Webpack version support dropping, we are using Webpack 5 so it should be good and I have tested the build locally

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -10,7 +10,6 @@ module.exports = {
   },
   output: {
     path: path.resolve(__dirname, "sphinx_wagtail_theme/static/dist"),
-    publicPath: "",
```
Comment (Stormheg) [inline webpack.config.js:13]: **Question:** why did you remove this option? Will removing it have adverse effects?

**Comment by Stormheg** [comment]
Not sure what is wrong with the Vercel deployment. I don't have access and cannot view the logs.

File: webpack.config.js (hunk 1 above)
Comment (lb-) [reply webpack.config.js:13]: Sorry. I should have called this out. The upgrade for css extract recommended removing this as it's no longer needed. 

Webpack will default this to `auto` instead. 

https://webpack.js.org/guides/public-path/
https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/CHANGELOG.md#200-2021-06-30

Comment (Stormheg) [reply webpack.config.js:13]: Thanks! Sounds fine to me.

**Comment by Stormheg** [comment]
Per https://github.com/wagtail/sphinx-wagtail-theme/issues/241#issuecomment-1635417602, preview deployments with Vercel have been broken for a while now. Not related to this PR.
